konnectivity-client/pkg/client/client.go

```diff
@@ -183,7 +186,7 @@ func (t *grpcTunnel) DialContext(ctx context.Context, protocol, address string)
 	}
 
 	random := rand.Int63() /* #nosec G404 */
-	resCh := make(chan dialResult, 1)
+	resCh := make(chan dialResult)
```

Comment:
I think I'd at least add a comment here. This channel was unbuffered before, but was made buffered by this commit: 9bbe1fc5fac157b660fc8a47e10d57c0ff29573c. This probably would not have happened if we at least had a comment there.

(the logic to return if sending to the channel blocks didn't exist back then, but was introduced here: 9f0e43ed. It probably was buggy before this commit too, as it would block indefinitely in that case)

```suggestion
        // This channel MUST NOT be buffered. The sender needs to know when we are not receiving things, so they can abort.
	resCh := make(chan dialResult)
```

What do you think?